src/Draco.Compiler.Cli/Program.cs

```diff
@@ -1,23 +1,88 @@
 using System;
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.IO;
 using System.Linq;
+using System.Threading.Tasks;
 using Draco.Compiler.Api.Scripting;
 using Draco.Compiler.Api.Syntax;
 
 namespace Draco.Compiler.Cli;
 
 internal class Program
 {
-    internal static void Main(string[] args)
+    internal static int Main(string[] args)
     {
-        var tree = ParseTree.Parse(""""
-            func main() {
-                var a = """
-                """;
-            }
-            """");
-        foreach (var diag in tree.GetAllDiagnostics())
-        {
-            Console.WriteLine(diag);
+        var fileArgument = new Argument<FileInfo>("--file", description: "Draco file");
+        var emitCSOption = new Option<bool>("--emit-cs", () => false, description: "Specifies if generated c# code should be saved to the disk");
+
+        var runCommand = new Command("run", "runs specified draco file")
+        {
+            fileArgument,
+            emitCSOption,
+        };
+        runCommand.SetHandler((file, emitCS) =>
+        {
+            Run(file, emitCS);
+        }, fileArgument, emitCSOption);
+
+        var generateParseTreeCommand = new Command("parse", "generates parse tree from specified draco file")
+        {
+            fileArgument,
+        };
+        generateParseTreeCommand.SetHandler((file) =>
+        {
+            GenerateParseTree(file);
+        }, fileArgument);
+
+
+        var generateCSCommand = new Command("codegen", "generates c# from specified draco file and displays it to the console")
+        {
+            fileArgument,
+            emitCSOption,
+        };
+        generateCSCommand.SetHandler((file, emitCS) =>
+        {
+            GenerateCSharp(file, emitCS);
+        }, fileArgument, emitCSOption);
+
+        var generateExeCommand = new Command("compile", "generates executable from specified draco file")
+        {
+            fileArgument,
+            emitCSOption
+        };
+        generateExeCommand.SetHandler((file, emitCS) =>
+        {
+            GenerateExe(file, emitCS);
+        }, fileArgument, emitCSOption);
+
+        var rootCommand = new RootCommand("CLI for the draco compiler");
```

Comment:
Is there no overload to add the below commands immediately? I don't know the API